### ISSUE TITLE
Debian 9 PHP 7 Uncaught Exception

### ISSUE THREAD

**Issue by MichalMMac** [fusiondirectory/fusiondirectory#33]
Hello

I am trying to setup Fusion Directory on Debian 9 with Apache 2.4 and PHP 7 and OpenLDAP 2.4.
I follow docs -> https://documentation.fusiondirectory.org/en/documentation/admin_installation
Version is 1.0.19 installed through default Debian apt repo (same for fusion directory schema)

I go through web setup withou any issues but when I try to log in with `fd-admin` account I get following errors:

Apache:
```
  Feb 02 17:58:11 smalldebian apache2[27017]: FusionDirectory [unauthenticated]: (view) error: PHP error: Only variables should be passed by reference (/usr/share/fusiondirectory/include/functions.inc, line 589)
    Feb 02 17:58:11 smalldebian slapd[1844]: <= mdb_equality_candidates: (roleOccupant) not indexed
    Feb 02 17:58:11 smalldebian apache2[27017]: FusionDirectory [fd-admin]: (view) error: PHP error: Only variables should be passed by reference (/usr/share/fusiondirectory/include/functions.inc, line 2538)
    Feb 02 17:58:11 smalldebian apache2[27017]: FusionDirectory [fd-admin]: (security) login: User "fd-admin" logged in successfully.
```

Web Page:
```
Fatal error: Uncaught Exception: Error: lexing failed because a rule matched an empty string. Input "0)} ... state TAGBODY in /usr/share/php/smarty3/sysplugins/smarty_internal_templatelexer.php:566 Stack trace: #0 /usr/share/php/smarty3/sysplugins/smarty_internal_templatelexer.php(263): Smarty_Internal_Templatelexer->yylex3() #1 /usr/share/php/smarty3/sysplugins/smarty_internal_smartytemplatecompiler.php(109): Smarty_Internal_Templatelexer->yylex() #2 /usr/share/php/smarty3/sysplugins/smarty_internal_templatecompilerbase.php(404): Smarty_Internal_SmartyTemplateCompiler->doCompile('<!-- Headline -...', true) #3 /usr/share/php/smarty3/sysplugins/smarty_internal_templatecompilerbase.php(335): Smarty_Internal_TemplateCompilerBase->compileTemplateSource(Object(Smarty_Internal_Template), false, NULL) #4 /usr/share/php/smarty3/sysplugins/smarty_template_compiled.php(199): Smarty_Internal_TemplateCompilerBase->compileTemplate(Object(Smarty_Internal_Template)) #5 /usr/share/php/smarty3/sysplugins/smarty_template_compiled.php(98) in /usr/share/php/smarty3/sysplugins/smarty_internal_templatelexer.php on line 566
```

Changelog says Fusion Directory is ready for PHP 7.
Any ideas what might went wrong?

**Comment by bilbo-the-hobbit** [comment]
Hello,

seems strange, can you paste here the fusiondirectory version from apt, smarty version, php ini if you changed something

Cheers

**Comment by MichalMMac** [comment]
```
Package: fusiondirectory                 
Version: 1.0.19-1
New: yes
State: installed
```

```
Package: smarty3                         
Version: 3.1.30-1
State: installed
```

php.ini should be default. It's Debian 9 in VM which was upgraded from Debian 8.
http://pastebin.com/cwUjsFqL

**Comment by MichalMMac** [comment]
`(view) error: PHP error: Only variables should be passed by reference (/usr/share/fusiondirectory/include/functions.inc, line 2538)` error are probably not relevant. When I got rid of them by editing code, lexing error is stil there.

**Comment by MCMic** [comment]
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=847571
The bug is in smarty-lexer it seems.

**Comment by MichalMMac** [comment]
Thanks! Good to know :-)

**Comment by bilbo-the-hobbit** [comment]
hello,

you can test our fixes release for the Only variables should be passed by reference

https://documentation.fusiondirectory.org/en/documentation_testing#testing-with-the-fixes-versions

Cheers

**Comment by MichalMMac** [comment]
smarty3 updated to `3.1.31`. No change.
From linked bugreport thread I understand bug has to be fixed in smarty-lexer which is used to generate some code for smarty3 and than we wait for new smarty3 version or install from source.

**Comment by MCMic** [comment]
Yes exactly, a bug in smarty-lexer create a bug in smarty while building the package.

**Comment by bilbo-the-hobbit** [comment]
Hello,

smarty-lexer has been fixed in debian stretch and our code has been adapted also should be fixes in 1.0.20
